src/dag.jl

```diff
@@ -31,6 +31,7 @@ function DAG{T}(sz = typemax(Int)) where T
     cond_push  = Condition()
     cond_empty = Condition()
     _buffer    = Set{Int}()
+    sz == Inf || sizehint!(_buffer, sz)
```

Comment:
@maltezfaria Is this still needed?

Note that because of line 26, `sz` is never `Inf` but could be `typemax(Int)`. And I would expect `sizehint!(_buffer, typemax(Int))` to exhaust memory...

Reply:
@ffevotte you are right, this is neither needed nor correct. @Vincent-Sivadon can you fix this in the PR?